driver-async/src/main/com/mongodb/async/client/ClientSessionImpl.java

```diff
@@ -74,27 +70,34 @@ public void startTransaction(final TransactionOptions transactionOptions) {
 
     @Override
     public void commitTransaction(final SingleResultCallback<Void> callback) {
-        if (!inTransaction) {
+        if (!canCommitOrAbort()) {
             throw new IllegalStateException("There is no transaction started");
         }
-        endTransaction(new CommitTransactionOperation(transactionOptions.getWriteConcern()), callback);
+        if (getServerSession().getStatementId() == 0) {
+            cleanupTransaction();
+            callback.onResult(null, null);
+        } else {
+            ReadConcern readConcern = transactionOptions.getReadConcern();
+            if (readConcern == null) {
+                throw new MongoInternalException("Invariant violated.  Transaction options read concern can not be null");
+            }
+            executor.execute(new CommitTransactionOperation(transactionOptions.getWriteConcern()),
+                    ReadPreference.primary(), readConcern, this,
+                    new SingleResultCallback<Void>() {
+                        @Override
+                        public void onResult(final Void result, final Throwable t) {
+                            cleanupTransaction();
+                            callback.onResult(result, t);
```

Comment:
Should have run the tests again before pushing a commit.  :(  This should be 

     callback.onResult(null, null);

to match the previous (and expected) behavior.